packages/client/src/components/Filters/Filters.tsx

```diff
@@ -98,10 +100,11 @@ const CheckboxList = ({ listLabel, filters, filtersName, selectedCheckboxes }: C
 };
 
 const Filters = () => {
-  const [checked, setChecked] = useState<{ name: string | null; value: string | null }[]>([]);
+  const [checked, setChecked] = useState<SingleFilterType[]>([]);
+  const { addFilters } = useFiltersContext();
 
-  const handleChecked = (e: ChangeEvent<HTMLInputElement>) => {
-    let updatedList = [...checked];
+  const handleChecked = (e: React.ChangeEvent<HTMLInputElement> & React.MouseEvent<HTMLDivElement>): void => {
```

Comment:
Ja bym zamienił to na:
```suggestion
  const handleChecked: React.FormEventHandler<HTMLDivElement> = (e) => {
```
P.S. [tutaj ](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/forms_and_events/) mała ściąga jak typować w takich sytuacjach.